proposal.rst

```diff
@@ -180,39 +180,57 @@ Motivations
 
 .. For example, think about the likely user-base, what they currently have to
 .. do and how your project will improve things for them.
 
-Project I want to do assumes that there will be tool for evaluating Letor on different datasets and compare its performance, so it useful for users wich want to understand Xapian abilities.
+Project I want to do assumes that there will be tool for evaluating Letor on different datasets and compare its performance. If it is completed successfully, there free suite with several ranking algorithms and features will available for users. It is useful not only for future developers, but for participants IR contests like TREC. Also it is possible to implement automated configuration choose using score classes.  
 
 Project Details
 ---------------
 
-.. Please go into plenty of detail in this section.
-
 **Describe any existing work and concepts on which your project is based.**
 
-There are several ranking algorithms with pretty same api implemented in Letor. So I think manager class should be developed to evaluate ranking and logging results. It will better to integrate it with testing classes, because it makes possible to develop suite for perfomance rating, wich will useful not only for ranking, but for all test cases. 
+There are several ranking algorithms with pretty same api implemented in Letor. So I think manager class should be developed to evaluate ranking and logging results. It will better to integrate it with Letor classes. Additional task is to improve perftest suite by running it on real data, wich will useful not only for ranking benchmark, but for all test cases. More detailed:
+
+* Implement class for preparing data, i.e. download it and convert to suitable form. The most convenient dataset is `FIRE <http://fire.irsi.res.in/fire/static/data>`_ and `TREC <https://trec.nist.gov/data/t2002_filtering.html>`_. It is hard to fully generalize this case, so I prefer to implement parser for each dataset:
+
+	* FIRE queries stored in inappropriate format, they should be transformed to use them in Xapian; It will possible to use English language corpus, because Eglish stemmer exists in Xapian.
+
+	* TREC dataset has big enough amount of documents, it is possible to combine some of them during pasing to create more complex queries.
+
+* Implement evaluator. According to xapian-letor/tests/api_letor.cc it is no big difference in applying rankers to data, it should just apply algorithms and return scores, so it could be another one class in xapian/letor. It would be nice to be able to call it same way as make check.
+
+* Write suite for combining documents. TREC dataset has big corpus of small documents. To get more content small documents could be merged. Also queries and qrels have to be transformed. After that evaluation could be done using Xapian API logic operations for queries. Since it is not needed for every Letor run, it should be separate tool, like xapian-letor/bin/xapian-prepare-trainingfile.cc. 
```

Comment:
If you combine documents, are the qrels going to make any kind of sense? Is there some paper or technique you're proposing to transform the qrels? Also, since you subsequently talk about wikipedia (which has large as well as small documents), do you actually need to do this?

Reply:
And how do you decide which groups of documents to combine?

Just grouping documents arbitrarily (e.g the first 10 docs together, then the next 10 together, etc) would produce larger documents, but they won't be representative of what a real dataset with 10x larger documents would be like.

Even combining small documents on the same subject may not produce a realistic larger document.  Think what the concatenated text of 1000 tweets on the same subject would look like - nobody would write a (say) 30,000 word document which looked like that.

Unless you have a good technique for this, it'll be better to find a test dataset with a larger document size.